runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -4771,13 +4776,23 @@ reservationLockEnter(TR::Node *node, int32_t lwOffset, TR::CodeGenerator *cg, TR
                                  node, TR::MemoryReference::createWithDisplacement(cg, objReg, lwOffset & 0x0000FFFF, lockSize), monitorReg);
       }
 
-   doneLabel->setEndInternalControlFlow();
-   generateDepLabelInstruction(cg, TR::InstOpCode::label, node, doneLabel, conditions);
+   generateLabelInstruction(cg, TR::InstOpCode::label, node, doneLabel);
+#if JAVA_SPEC_VERSION >= 19
+      /*
+       * Adjust J9VMThread ownedMonitorCount for execution paths that do not
+       * go out of line to acquire the lock. It is safe and efficient to do
+       * this unconditionally. There is no need to check for overflow.
+       */
+      generateTrg1MemInstruction(cg, TR::InstOpCode::Op_load, node, tempReg, TR::MemoryReference::createWithDisplacement(cg, metaReg, fej9->thisThreadGetOwnedMonitorCountOffset(), TR::Compiler->om.sizeofReferenceAddress()));
+      generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addi, node, tempReg, tempReg, 0x1);
+      generateMemSrc1Instruction(cg, TR::InstOpCode::Op_st, node, TR::MemoryReference::createWithDisplacement(cg, metaReg, fej9->thisThreadGetOwnedMonitorCountOffset(), TR::Compiler->om.sizeofReferenceAddress()), tempReg);
+#endif //JAVA_SPEC_VERSION >= 19
 
    TR_PPCOutOfLineCodeSection *outlinedHelperCall = new (cg->trHeapMemory()) TR_PPCOutOfLineCodeSection(node, TR::call, NULL, callLabel, doneOOLLabel, cg);
    cg->getPPCOutOfLineCodeSectionList().push_front(outlinedHelperCall);
 
-   generateLabelInstruction(cg, TR::InstOpCode::label, node, doneOOLLabel);
+   generateDepLabelInstruction(cg, TR::InstOpCode::label, node, doneOOLLabel, conditions);
```

Comment:
done00LLable might reduce the code readability.  in later code, i saw callReturnLabel which sounds better and more readable.

Reply:
I can change this.